### ISSUE TITLE
Searching the manual does not return all relevant results

### ISSUE THREAD

**Issue by mardukbp** [wxMaxima-developers/wxmaxima#473]
I was looking for a command that extracts the coefficient of a term in an expression, so I wrote "coefficient" in the search box (not in the Index) of the Maxima manual. Since the search did not return a useful command, I asked in Maxima-discuss and I was told about the `coeff` command. This should have shown up when searching the manual.

**Comment by gunterkoenigsmann** [comment]
This means the manual should contain the literal word "coefficient" in this chapter. Doesn't even sound like a bad idea and it is always good to correct these things: The person who writes a manual will most probably not search in it. Unfortunately the manual with the maxima commands is maintained by the maxima project => will close this bug here. But feel free to report it at https://sourceforge.net/p/maxima/bugs/ or (that would be greatest) to grab a git client (for linux rabbitvcs is fine, for windows http://git-scm.com/ should provide you with one), clone the repo and to correct the source: There is a patch tracker at https://sourceforge.net/p/maxima/patches/ and help is always welcome.

Kind regards,
Gunter.

**Comment by mardukbp** [comment]
Not so fast. The manual does contain the word "coefficient":

>
Function: coeff  
    coeff (expr, x, n)  
    coeff (expr, x) 

> Returns the coefficient of x^n in expr, where expr is a polynomial or a monomial term in x.

**Comment by gunterkoenigsmann** [comment]
Then I need more information:

What operating system are you working on? And how did you search in the manual?

**Comment by mardukbp** [comment]
OS: Arch Linux

I searched pressing `F1` and then clicked on the `Search` tab and entered "coefficient".

**Comment by gunterkoenigsmann** [comment]
Weird: I use ubuntu but I think both ubuntu and arch linux use yelp as 
the help browser.
On my system the help browser shows two manuals:
 - The manual for wxMaxima being the graphical user interface and
 - The manual for maxima that does the actual mathematics.
If I search in the manual for maxima for "coefficient" I get many 
search results.

**Comment by mardukbp** [comment]
In my system the help browser only shows the maxima manual. Searching for "coefficient" returns several results which do not include the command `coeff`.

**Comment by andrejv** [comment]
The help browser is provided by the wxwidgets library. This is probably an issue there.

**Comment by mardukbp** [comment]
Ok, I did some research and I have a better idea of what is happening. The results obtained using the search tab only contain the title of the sections of the Maxima documentation (i.e. the HTML files) that include the keyword of interest. 

That IMHO is either a shortcoming of the wxWidgets HTML help browser or of wxMaxima's use thereof. The search results should include an entry to each function whose documentation includes the keyword searched. Just like the entries in the index tab. The goal should be to have an inverse dictionary that tells you which command does what you want.

**Comment by andrejv** [comment]
There are two tabs in the browser. There is an "Index" from which you can search through the index (maxima commands and special variable names) and a "Search" tab which should search through all text. Thats how it works on OS X.

**Comment by mardukbp** [comment]
> a "Search" tab which should search through all text

That is indeed what happens. The problem is that the results show the titles of the sections that contain the keyword searched. Not very practical considering the length of the HTML pages and very annoying when you know that every function has an HTML anchor.

I have been thinking that it would be better to have a keyboard shortcut that when activated splits the window vertically showing a pane in the bottom (like the IPython Notebook). This pane should have a search box and the search results should be a list where each item contains the first lines in the documentation of the function, e.g. searching for **coefficient** should yield:

```
Function: coeff  
    coeff (expr, x, n)  
    coeff (expr, x) 

Returns the coefficient of x^n in expr, where expr is a polynomial or a monomial term in x. 
```

And so on for other functions that contain the keyword searched in their description. This has the advantage of also showing how to call the function of interest. When clicking on a result item the help browser should pop up showing the complete documentation for the function.

Inspired by the IPython Notebook this pane should also be activated when pressing tab when the cursor is at the end of the name of a built-in Maxima function. It would be great to have this functionality for user-defined functions too.

I believe that it would be easier to write a script that extracts the first lines in the definition of the functions and gathers all these snippets in a single file. This would have to be done only once outside wxMaxima.

**Comment by gunterkoenigsmann** [comment]
The Right Place to do this would probably convincing the Maxima project that they should omit the --split options for their texi2html calls. But as both a monolitic documentation (that can be searched easily) and a documentation that is split into chapters that are easy to handle have their advantages I am not sure if that is possible.

**Comment by mardukbp** [comment]
I just checked the `doc` dir in the Maxima repo. The `--split` calls are done in order to have each chapter in a separate HTML file. I do not understand why omitting them would alleviate the problem. The current format of the documentation is useful as a reference manual but not for a quick look up. It would be better if they used a script to split the headers of the commands, like I showed in my comment above, from the body and combine all headers in a single file.
